engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/HashCodeNode.java

```diff
@@ -268,12 +267,12 @@ long hashCodeForAtomWithCustomComparator(
     var args = new Object[] { cachedComparator, atom};
     var result = invokeFunctionNode.execute(compareMethod, null, State.create(ctx), args);
     if (!interop.isNumber(result)) {
-      throw new PanicException("Custom comparator must return a number", this);
+      throw ctx.raiseAssertionPanic(this, "Custom comparator must return a number", null);
     } else {
       try {
         return interop.asLong(result);
       } catch (UnsupportedMessageException e) {
-        throw new IllegalStateException(e);
+        throw EnsoContext.get(this).raiseAssertionPanic(this, null, e);
```

Comment:
```suggestion
        throw ctx.raiseAssertionPanic(this, null, e);
```
why not ? (Also in other places)